lib/concurrent/configuration.rb

```diff
@@ -62,17 +31,48 @@ def self.create_stdlib_logger(level = Logger::FATAL, output = $stderr)
              progname,
              formatted_message
     end
-    logger
+
+    lambda do |level, progname, message = nil, &block|
+      logger.add level, message, progname, &block
+    end
   end
 
   # Use logger created by #create_stdlib_logger to log concurrent-ruby messages.
   def self.use_stdlib_logger(level = Logger::FATAL, output = $stderr)
-    logger = create_stdlib_logger level, output
-    Concurrent.global_logger = lambda do |level, progname, message = nil, &block|
-      logger.add level, message, progname, &block
-    end
+    Concurrent.global_logger = create_stdlib_logger level, output
   end
 
+  # Suppresses all output when used for logging.
+  NULL_LOGGER   = lambda { |level, progname, message = nil, &block| }
+
+  # @!visibility private
+  GLOBAL_LOGGER = AtomicReference.new(create_stdlib_logger(Logger::WARN))
```

Comment:
Why the change from `NULL_LOGGER`? Logging is a side-effect. In general, I prefer not to create unnecessary side-effects and, instead, let the user decide for themselves if they want to log.